agent/engine/docker_container_engine.go

```diff
@@ -848,3 +857,27 @@ func (dg *dockerGoClient) removeImage(imageName string) error {
 	}
 	return client.RemoveImage(imageName)
 }
+
+// LoadImage invokes github.com/fsouza/go-dockerclient.Client's
+// LoadImage API with a timeout
+func (dg *dockerGoClient) LoadImage(opts docker.LoadImageOptions, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
```

Comment:
You may want to pass this `ctx` into the `opts` since `LoadImage`, so that docker will cancel the request to `LoadImage`.